compliance/collection/auditlog/auditlog_impl.go

```diff
@@ -53,29 +48,34 @@ func (s *auditLogReaderImpl) StartReader(ctx context.Context) (bool, error) {
 	if err != nil {
 		if os.IsNotExist(err) {
 			log.Errorf("Audit log file %s doesn't exist on this compliance node", s.logPath)
-			return false, nil
+			err = nil
+		} else {
+			// handle other errors
+			log.Errorf("Failed to open file: %v", err)
 		}
-		// handle other errors
-		log.Errorf("Failed to open file: %v", err)
+		s.stopper.Flow().StopWithError(err)
+		s.stopper.Flow().ReportStopped()
 		return false, err
 	}
 
 	go s.readAndForwardAuditLogs(ctx, t)
 	return true, nil
 }
 
-func (s *auditLogReaderImpl) StopReader() bool {
-	return s.stopC.Signal()
+func (s *auditLogReaderImpl) StopReader() {
+	s.stopper.Client().Stop()
+	_ = s.stopper.Client().Stopped().Wait()
```

Comment:
Absence of this line lead to the following error appear from time to time in logs when running tests.
```
collection/auditlog: 2022/11/30 12:19:56.751189 auditlog_impl.go:158: Error: Audit log tailer stopped with an error no such file or directory
```

Looks like I fixed a bug.